mambaSharedFramework/HLSValueTypes.swift

```diff
@@ -266,4 +454,68 @@ public func ==(lhs: HLSCodecArray, rhs: HLSCodecArray) -> Bool {
     return lhs.codecs == rhs.codecs
 }
 
+/// Represents information to assist in view presentation.
+///
+/// Indicates when video content in the Variant Stream requires specialized rendering to be properly displayed.
+public struct HLSVideoLayout: Equatable, FailableStringLiteralConvertible {
+    /// Each specifier controls one aspect of the entry. That is, the specifiers are disjoint and the values for a
+    /// specifier are mutually exclusive.
+    public let layouts: [VideoLayout]
+    /// The client SHOULD assume that the order of entries reflects the most common presentation in the content.
+    ///
+    /// For example, if the content is predominantly stereoscopic, with some brief sections that are monoscopic then the
+    /// Multivariant Playlist SHOULD specify `REQ-VIDEO-LAYOUT="CH-STEREO,CH-MONO"`. On the other hand, if the content
+    /// is predominantly monoscopic then the Multivariant Playlist SHOULD specify `REQ-VIDEO-LAYOUT="CH-MONO,CH-STEREO"`.
+    public let predominantLayout: VideoLayout
+
+    public enum VideoLayout: String {
+        /// Monoscopic.
+        ///
+        /// Indicates that a single image is present.
+        case chMono = "CH-MONO"
+        /// Stereoscopic.
+        ///
+        /// Indicates that both left and right eye images are present.
+        case chStereo = "CH-STEREO"
+
+        init?(str: Substring) {
+            switch str {
+            case "CH-MONO": self = .chMono
+            case "CH-STEREO": self = .chStereo
+            default: return nil
+            }
+        }
+    }
+
+    public init?(string: String) {
+        var layouts = [VideoLayout]()
+        for str in string.split(separator: ",") {
+            if let layout = VideoLayout(str: str) {
+                layouts.append(layout)
+            } else {
+                // Favor failing to parse the whole array if we find an unrecognized layout, so that we don't risk mis-
+                // reporting the existing layouts.
+                return nil
+            }
+        }
+        guard let firstLayout = layouts.first else {
+            return nil
+        }
+        self.predominantLayout = firstLayout
+        self.layouts = layouts
+    }
+
+    public init?(layouts: [VideoLayout]) {
+        guard let predominantLayout = layouts.first else { return nil }
+        self.layouts = layouts
+        self.predominantLayout = predominantLayout
+    }
+
+    public func containsStereo() -> Bool {
+        layouts.contains(.chStereo)
+    }
 
+    public func containsMono() -> Bool {
+        layouts.contains(.chMono)
+    }
```

Comment:
not sure if it makes sense but subscripts could be used here for these `contains` functions
```swift
    subscript(layout: VideoLayout) -> Bool {
        layouts.contains(layout)
    }
```

Reply:
Interesting... Again, I was kind of keeping with the existing style, as I saw it in the `HLSCodecArray`... But I can look to have a nicer syntax 👍

Reply:
I wasn't such a fan of the `subscript` approach, as I'm not sure the following is _totally_ clear what it is implying:
```swift
if layout[.chMono] {
    // Do something if the layout has mono
}
```
So instead, I made a convenience that mirrors the array `contains` method instead here: 440593faa16ffa74b52799acc8a957c5f6eaaf9b

So more like:
```swift
if layout.contains(.chMono) {
    // Do something if the layout has mono
}
```

What do you think?

Reply:
Yeah I like this, seems like a "best of all worlds" solution using the enum's and it won't require a unique method if a new enum case is added.